src/notebook/model.ts

```diff
@@ -482,12 +449,12 @@ class NotebookModel implements INotebookModel {
   /**
    * Handle a change to a cell state.
    */
-  private _onCellStateChanged(cell: ICellModel, change: IChangedArgs<ICellModel>): void {
+  private _onCellStateChanged(cell: ICellModel, change: IChangedArgs<any>): void {
     if (change.name === 'dirty' && change.newValue) {
       this.dirty = true;
     }
-    if (change.name === 'focused') {
-      if (change.newValue) {
+    if (change.name === 'mode') {
+      if (change.newValue === 'edit') {
```

Comment:
How about combining these so you don't need to indent twice:

`if (change.name === 'mode' && change.newValue === 'edit') { ... }`